src/main/java/net/snowflake/client/log/JDK14Logger.java

```diff
@@ -134,7 +134,12 @@ private void logInternal(Level level, String msg, boolean masked) {
   private void logInternal(Level level, String msg, Object... arguments) {
     if (jdkLogger.isLoggable(level)) {
       String[] source = findSourceInStack();
-      String message = MessageFormat.format(refactorString(msg), evaluateLambdaArgs(arguments));
+      String message = "";
+      try {
+        message = MessageFormat.format(refactorString(msg), evaluateLambdaArgs(arguments));
+      } catch (IllegalArgumentException e) {
+        message = "Unable to format msg: " + msg;
```

Comment:
minor request: Could we also add the exception message string to the altered message string instead of discarding the exception information entirely? There can be multiple causes to this throw so a capture of `ex.getMessage()` could help when an investigation is ever required.

Reply:
should we want us to block release on this recommendation or we can fix this later 
@sfc-gh-igarish @sfc-gh-skumbham  comments?